editor/editor_asset_installer.cpp

```diff
@@ -338,26 +367,102 @@ String EditorAssetInstaller::get_asset_name() const {
+					extension_icon_map["cs"] = get_editor_theme_icon(SNAME("CSharpScript"));
+				} else {
+					// Mark C# support as unavailable.
+					extension_icon_map["cs"] = get_editor_theme_icon(SNAME("ImportFail"));
+				}
+
+				extension_icon_map["res"] = get_editor_theme_icon(SNAME("Resource"));
+				extension_icon_map["tres"] = get_editor_theme_icon(SNAME("Resource"));
+				extension_icon_map["atlastex"] = get_editor_theme_icon(SNAME("AtlasTexture"));
+				// By default, OBJ files are imported as Mesh resources rather than PackedScenes.
+				extension_icon_map["obj"] = get_editor_theme_icon(SNAME("Mesh"));
+
+				extension_icon_map["txt"] = get_editor_theme_icon(SNAME("TextFile"));
+				extension_icon_map["md"] = get_editor_theme_icon(SNAME("TextFile"));
+				extension_icon_map["rst"] = get_editor_theme_icon(SNAME("TextFile"));
+				extension_icon_map["json"] = get_editor_theme_icon(SNAME("TextFile"));
+				extension_icon_map["yml"] = get_editor_theme_icon(SNAME("TextFile"));
+				extension_icon_map["yaml"] = get_editor_theme_icon(SNAME("TextFile"));
+				extension_icon_map["toml"] = get_editor_theme_icon(SNAME("TextFile"));
+				extension_icon_map["cfg"] = get_editor_theme_icon(SNAME("TextFile"));
+				extension_icon_map["ini"] = get_editor_theme_icon(SNAME("TextFile"));
+			}
+		} break;
+	}
+}
+
 void EditorAssetInstaller::_bind_methods() {
 }
 
 EditorAssetInstaller::EditorAssetInstaller() {
 	VBoxContainer *vb = memnew(VBoxContainer);
 	add_child(vb);
 
-	asset_contents = memnew(Label);
-	vb->add_child(asset_contents);
+	HBoxContainer *asset_status = memnew(HBoxContainer);
+	vb->add_child(asset_status);
+
+	Label *asset_label = memnew(Label);
+	asset_label->set_text(TTR("Contents of asset:"));
+	asset_status->add_child(asset_label);
+
+	asset_title_label = memnew(Label);
+	asset_title_label->set_theme_type_variation("HeaderSmall");
+	asset_status->add_child(asset_title_label);
+
+	asset_status->add_spacer();
+
+	asset_conflicts_label = memnew(Label);
+	asset_conflicts_label->set_theme_type_variation("HeaderSmall");
+	asset_status->add_child(asset_conflicts_label);
+
+	skip_toplevel_check = memnew(CheckBox);
+	skip_toplevel_check->set_text(TTR("Ignore the root directory when extracting files"));
+	skip_toplevel_check->set_h_size_flags(Control::SIZE_SHRINK_BEGIN);
+	skip_toplevel_check->connect("toggled", callable_mp(this, &EditorAssetInstaller::_set_skip_toplevel));
+	vb->add_child(skip_toplevel_check);
 
 	tree = memnew(Tree);
 	tree->set_v_size_flags(Control::SIZE_EXPAND_FILL);
-	tree->connect("item_edited", callable_mp(this, &EditorAssetInstaller::_item_edited));
+	tree->connect("item_edited", callable_mp(this, &EditorAssetInstaller::_item_checked));
 	tree->connect("check_propagated_to_item", callable_mp(this, &EditorAssetInstaller::_check_propagated_to_item));
 	vb->add_child(tree);
 
-	error = memnew(AcceptDialog);
-	add_child(error);
+	set_title(TTR("Select asset files to install"));
```

Comment:
As much as I personally dislike it, our convention for titles is to use Title Case.
```suggestion
	set_title(TTR("Select Asset Files to Install"));
```